packages/headless/src/features/commerce/product-templates/product-templates-helpers.ts

```diff
@@ -105,8 +150,88 @@ const getFieldValuesFromProduct = (
   return isArray(rawValue) ? rawValue : [rawValue];
 };
 
+const getFirstValidStringFieldValueFromProduct = (
+  fieldNames: string[],
+  product: Product
+) => {
+  const rawValues = fieldNames.map((fieldName) =>
+    getProductProperty(product, fieldName)
+  );
+
+  const value = rawValues.find((value) => typeof value === 'string');
+
+  if (value === undefined) {
+    return value;
+  }
+
+  return value as string;
+};
+
+const getCouldNotRetrievePropertyWarning = (
+  property: string,
+  lookupFields: string[]
+) =>
+  `'${property}' could not be retrieved from field${lookupFields.length > 1 ? 's' : ''} '${lookupFields.join("', '")}'.`;
+
+const getFirstValidNumberFieldValueFromProduct = (
+  fieldNames: string[],
+  product: Product
+) => {
+  const rawValues = fieldNames.map((fieldName) =>
+    getProductProperty(product, fieldName)
+  );
+
+  const value = rawValues.find((value) => typeof value === 'number');
+
+  if (value === undefined) {
+    return value;
+  }
+
+  return value as number;
+};
+
+export const getProductNameFromStandardLookupFields = (product: Product) => {
+  const lookupFields = ['ec_name', 'ec_product_id', 'permanentid'];
```

Comment:
At the moment I'm using a hardcoded, opinionated list of lookup values that will be verified in the specified order (i.e., in this case if ec_name isn't defined on a product, or if its value isn't a string, then we will fallback to ec_product_id, and then to permanentid).

Eventually, customers might come up with valid use cases for specifying their own lookup fields (which would probably be set in additionalFields). If this happens, it will be straightforward to add options on the getRequiredProductPropertiesForAnalytics helper function.

We could even image cases where, for instance, determining the field to use for the price could be done by a function rather than a simple list of prioritized lookup fields. Again, having a single function to get the required product properties for analytics allows us to add support for this as needed.

For now, though, I think we should keep it simple 😁